### PR TITLE
Autoscaling Workflow Enhancement - Part 3 

### PR DESCRIPTION
**Pull request by kr11**
## Pull Request Description

This PR updates the PodAutoscaler workflow, enable to update the pod replicas based on the KPA algorithm. 

In addition, it replaces the Deployment structure with Unstructured to increase flexibility. 

For more details on why we're not using `autoscalingv1.Scale`, see issue #102 

The README has been updated to demonstrate these changes.

## Related Issues
Resolves: #102 

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

File: pkg/controller/podautoscaler/podautoscaler_controller.go [hunk 1]
```diff
@@ -233,7 +229,17 @@ func (r *PodAutoscalerReconciler) reconcileKPA(ctx context.Context, pa autoscali
 	setCondition(&pa, "AbleToScale", metav1.ConditionTrue, "SucceededGetScale", "the HPA controller was able to get the target's current scale")
 
 	// current scale's replica count
-	currentReplicas := *scale.Spec.Replicas
+	currentReplicasInt64, found, err := unstructured.NestedInt64(scale.Object, "spec", "replicas")
+	if err != nil || !found {
```
Comment (Jeffwan) [inline pkg/controller/podautoscaler/podautoscaler_controller.go]: em, if all of them returns. why not use two separate code blocks?

File: pkg/controller/podautoscaler/podautoscaler_controller.go [hunk 2]
```diff
@@ -445,26 +448,48 @@ func (r *PodAutoscalerReconciler) updateStatus(ctx context.Context, pa *autoscal
 // It may return both valid metricDesiredReplicas and an error,
 // when some metrics still work and HPA should perform scaling based on them.
 // If PodAutoscaler cannot do anything due to error, it returns -1 in metricDesiredReplicas as a failure signal.
-func (r *PodAutoscalerReconciler) computeReplicasForMetrics(ctx context.Context, pa autoscalingv1alpha1.PodAutoscaler, scale *appsv1.Deployment) (replicas int32, metrics string, timestamp time.Time, err error) {
+func (r *PodAutoscalerReconciler) computeReplicasForMetrics(logger klog.Logger, ctx context.Context, pa autoscalingv1alpha1.PodAutoscaler, scale *unstructured.Unstructured) (replicas int32, metrics string, timestamp time.Time, err error) {
```
Comment (Jeffwan) [inline pkg/controller/podautoscaler/podautoscaler_controller.go]: do we need to pass the logger as argument? can we try `logger := klog.FromContext(ctx)`?

Comment (kr11) [reply pkg/controller/podautoscaler/podautoscaler_controller.go]: Great! I have fix it.

File: pkg/controller/podautoscaler/podautoscaler_controller.go (hunk 1 above)
Comment (kr11) [reply pkg/controller/podautoscaler/podautoscaler_controller.go]: > em, if all of them returns. why not use two separate code blocks?

Sure, I have separate it, please review the modified code.

**Pull request merged**